rcgen/src/lib.rs

```diff
@@ -331,21 +342,35 @@ impl DistinguishedName {
 	/// let mut dn = DistinguishedName::new();
 	/// dn.push(DnType::OrganizationName, "Crab widgits SE");
 	/// dn.push(DnType::CommonName, DnValue::PrintableString("Master Cert".try_into().unwrap()));
-	/// assert_eq!(dn.get(&DnType::OrganizationName), Some(&DnValue::Utf8String("Crab widgits SE".to_string())));
-	/// assert_eq!(dn.get(&DnType::CommonName), Some(&DnValue::PrintableString("Master Cert".try_into().unwrap())));
+	/// assert_eq!(dn.get(&DnType::OrganizationName).get(0), Some(&DnValue::Utf8String("Crab widgits SE".to_string())).as_ref());
+	/// assert_eq!(dn.get(&DnType::CommonName).get(0), Some(&DnValue::PrintableString("Master Cert".try_into().unwrap())).as_ref());
 	/// ```
 	pub fn push(&mut self, ty: DnType, s: impl Into<DnValue>) {
-		if !self.entries.contains_key(&ty) {
-			self.order.push(ty.clone());
-		}
-		self.entries.insert(ty, s.into());
+		self.entries.push_front((ty, s.into()));
 	}
-	/// Iterate over the entries
-	pub fn iter(&self) -> DistinguishedNameIterator<'_> {
```

Comment:
tbh I don't understand why the previous implementation would require the `DistinguishedNameIterator` as own iterator implementation. 

I'm implementing `Iterator` for `DistinguishedName` - so `iter()` is actually still available even though the explicit struct function was removed. 
I would argue that the interface is actually not breaking - `iter()` is still available after the change. 

@cpu  Is there a better way to achieve this so that semver accepts the change as non-breaking?

Reply:
@da-kami implementing `Iterator for DistinguishedName` does not make any method called `iter()` available -- `Iterator::iter()` does not exist. If you want to make it possible to represent multiple entries with the same `DnType`, I'd suggest changing the representation to hold a `Vec<(DnType, DnValue)>` and then build on top of that.